data/trie/patriciaMerkleTrie.go

```diff
@@ -515,6 +556,24 @@ func (tr *patriciaMerkleTrie) Snapshot() error {
 	return nil
 }
 
+func (tr *patriciaMerkleTrie) newSnapshotDb() (data.DBWriteCacher, error) {
+	db, err := storageUnit.NewDB(
+		storageUnit.DBType(tr.snapshotDbCfg.Type),
+		path.Join(tr.snapshotDbCfg.FilePath, strconv.Itoa(tr.snapshotId)),
+		tr.snapshotDbCfg.BatchDelaySeconds,
+		tr.snapshotDbCfg.MaxBatchSize,
+		tr.snapshotDbCfg.MaxOpenFiles,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	tr.snapshotId++
```

Comment:
tr.snapshot is always 0 when restarting the node. Is this the intended behavior? It might overwrite previously saved checkpoint data and be unable to serve other peers with checkpoint trie nodes.

Reply:
Changed, so now before creating a db, we check if the path exists.